src/Compilers/CSharp/Portable/Binder/MethodArgumentInfo.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Information about the arguments of a call that can turned into a BoundCall later without recalculating
+    /// default arguments.
+    /// </summary>
+    internal sealed record MethodArgumentInfo(
```

Comment:
nit: I don't think we need a record here. We don't need value equality semantics. 
On the other hand, I understand it's a nice shorthand to avoid typing constructor... 
Since this is our first record usage, I'm curious what others think.

Reply:
I did think about that. It seemed like a nice place to introduce them to the compiler, but I won't cry if you all ask me to just use a regular class here instead.

---
In reply to: [549493565](https://github.com/dotnet/roslyn/pull/50105#discussion_r549493565) [](ancestors = 549493565)

Reply:
> I don't think we need a record here. We don't need value equality semantics. 

Feels like an overkill to me too.

---
In reply to: [549502386](https://github.com/dotnet/roslyn/pull/50105#discussion_r549502386) [](ancestors = 549502386,549493565)